lightning/src/ln/channelmanager.rs

```diff
@@ -5759,12 +5759,12 @@ where
 		});
 	}
 
-	fn get_relevant_txids(&self) -> Vec<Txid> {
+	fn get_relevant_txids(&self) -> Vec<(Txid, BlockHash)> {
 		let channel_state = self.channel_state.lock().unwrap();
 		let mut res = Vec::with_capacity(channel_state.short_to_chan_info.len());
 		for chan in channel_state.by_id.values() {
-			if let Some(funding_txo) = chan.get_funding_txo() {
-				res.push(funding_txo.txid);
+			if let (Some(funding_txo), Some(block_hash)) = (chan.get_funding_txo(), chan.get_funding_tx_confirmed_in()) {
+				res.push((funding_txo.txid, block_hash));
```

Comment:
Do we want to skip txids that don't have a corresponding block hash? Or are these guaranteed to both be either `Some` or `None`?

Reply:
Yes, I thought that we'd want to skip any non-previously confirmed transactions anyways, and if they are confirmed they would have the `funding_tx_confirmed_in` field set anyways?

Reply:
Yeah, exactly, was included in https://github.com/lightningdevkit/rust-lightning/pull/1796/commits/72e9e850294219c5a235fcb8a71e352790c36751.